### PR TITLE
[pybind11] Use EasyBlock to install also with pip

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

- [x] Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2115

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusi6543.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/36aae9c468b73c3e88444bea4cdeddc9 for a full test report.

**Comment by Flamefire** [comment]
Tested with scipy 1.4.1 which now uses the installed pybind11 instead of downloading it

File: easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/pybind11/__init__.py b/pybind11/__init__.py
+index c625e8c..5c2e3bd 100644
+--- a/pybind11/__init__.py
++++ b/pybind11/__init__.py
+@@ -27,10 +27,15 @@ def get_include(user=False):
+ 
+         dist_cobj = dist.get_command_obj('install', create=True)
+ 
++        # Very crude hack to get install prefix from <prefix>/lib/python3.7/site-packages/pybind11
++        prefix = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__)))))
```
Comment (smoors) [inline easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch:13]: @Flamefire not sure if it is necessary here, but `__file__` depends on the location from where it has been called, so maybe this is safer?
```suggestion
+        prefix = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))

```
also, maybe use pathlib instead?
```
from pathlib import Path
prefix = Path(__file__).resolve().parents[4]
```

File: easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch [hunk 2]
```diff
@@ -0,0 +1,23 @@
+The current function doesn't take into account that a custom install prefix might have been set
+at installation time which we do in EB.
+Insert a quick hack to make it work as it should (fixed in 2.5.0)
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch:3]: Ok, fixed the commit

File: easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/p/pybind11/pybind11-2.4.3-fix-get-include.patch]: Good suggestion, that looks better. Fixed the commit

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
taurusi5130.taurus.hrsk.tu-dresden.de - Linux RHEL 7.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1dc4528aefb61c247373c157f95c6a99 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node254.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2683 v4 @ 2.10GHz (broadwell), Python 2.7.5
See https://gist.github.com/d372d5adfad128676373d1e60b11994f for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
sms - Linux centos linux 7.6.1810, x86_64, AMD EPYC 7601 32-Core Processor (zen), Python 2.7.5
See https://gist.github.com/7b5df2de8b313da00f4052866cb3024a for a full test report.

**Comment by Flamefire** [comment]
Rebased to retrigger CI, no changes

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node362.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/8a9def46a21c317151352d43315a6325 for a full test report.

**Comment by smoors** [comment]
Going in, thanks @Flamefire!

**Pull request merged**